cluster/charts/crossplane/values.yaml

```diff
@@ -37,6 +37,10 @@ customLabels: {}
 customAnnotations: {}
 
 serviceAccount:
+  # -- Specifies whether Crossplane ServiceAccount should be created
+  create: true
+  # -- Provide the name of an already created Crossplane ServiceAccount. Required when `serviceAccount.create` is `false`
+  name: ""
```

Comment:
AFAIR, other chart do respect the name even if create is true, you can just remove the `if`s and default to the computed name if not provided, something like`{{ default .Values.serviceAccount.name (template "crossplane.name" .) }}` (not sure about the syntax, just going by heart, please double check it)